kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/balance/view/BalanceView.java

```diff
@@ -1,120 +1,260 @@
+					arrowFromX = arrow.getPivotX() - getResources().getDimensionPixelOffset(R.dimen.kinecosystem_main_medium_margin);
+					arrowToX = arrowFromX + arrow.getWidth();
+					arrow.getViewTreeObserver().removeOnGlobalLayoutListener(this);
+				}
+			}
+		});
+
+		this.subTitleTextView.setText(subtitle);
+	}
+
+	private void initColors() {
+		blueColor = ContextCompat.getColor(getContext(), R.color.kinecosystem_bluePrimary);
+		orangeColor = ContextCompat.getColor(getContext(), R.color.kinecosystem_orange);
+		redColor = ContextCompat.getColor(getContext(), R.color.kinecosystem_red);
+		subTitleColor = ContextCompat.getColor(getContext(), R.color.kinecosystem_subtitle);
+	}
+
+	private void initSizes() {
+		Resources resources = getResources();
+		balanceTextSize = resources.getDimension(R.dimen.kinecosystem_title_xlarge_size);
+		subTitleTextSize = resources.getDimension(R.dimen.kinecosystem_sub_title_size);
+	}
+
+
+	private void addBalanceViewFactory() {
+		balanceText.setFactory(new ViewFactory() {
+			@Override
+			public View makeView() {
+				TextView balanceText = new TextView(getContext());
+				balanceText.setTextSize(TypedValue.COMPLEX_UNIT_PX, balanceTextSize);
+				balanceText.setTextColor(blueColor);
+				return balanceText;
+			}
+		});
+	}
+
+	private void addSubTitleViewFactory() {
+		subTitleTextView.setFactory(new ViewFactory() {
+			@Override
+			public View makeView() {
+				TextView subTitle = new TextView(getContext());
+				subTitle.setTextSize(TypedValue.COMPLEX_UNIT_PX, subTitleTextSize);
+				subTitle.setTextColor(subTitleColor);
+				return subTitle;
+			}
+		});
+	}
+
+	@Override
+	public void animateArrow(boolean showArrow) {
+		if (showArrow) {
+			postOnAnimation(new Runnable() {
+				@Override
+				public void run() {
+					balanceText.animate().translationX(arrowFromX).setInterpolator(new LinearInterpolator()).setDuration(ANIM_DURATION).start();
+					arrow.animate().scaleX(FLOAT_1).translationX(arrowFromX).alpha(FLOAT_1).setInterpolator(new LinearInterpolator()).setDuration(ANIM_DURATION).start();
+				}
+			});
+		} else {
+			postOnAnimation(new Runnable() {
+				@Override
+				public void run() {
+					balanceText.animate().translationX(arrowToX).setInterpolator(new LinearInterpolator()).setDuration(ANIM_DURATION).start();
+					arrow.animate().scaleX(FLOAT_08).translationX(arrowToX).alpha(FLOAT_0).setInterpolator(new LinearInterpolator()).setDuration(ANIM_DURATION).start();
+				}
+			});
+		}
+
+	}
+
+	@Override
+	public void attachPresenter(IBalancePresenter presenter) {
+		this.balancePresenter = presenter;
+		this.balancePresenter.onAttach(this);
+	}
+
+	@Override
+	public void updateBalance(final String balance) {
+		if (this.balanceText != null) {
+			post(new Runnable() {
+				@Override
+				public void run() {
+					balanceText.setText(balance);
+				}
+			});
+		}
+	}
+
+	@Override
+	public void setWelcomeSubtitle() {
+		TextView nextTextView = (TextView) subTitleTextView.getNextView();
```

Comment:
UI thread?

Reply:
not need, it's only called on the main thread.